### DISCUSSION TITLE
Ganging multiple boards together

### DISCUSSION THREAD

**Discussion by JimDrewGH** [gusmanb/logicanalyzer#157] [Q&A] [answered]
So, I made a few of the new v6.0 boards (be aware that the gerber data for these boards did NOT include the pin labeling or component IDs in the top silkscreen!).  Now that I have them I can not figure out how to enable using 2 boards together.  My sole purpose for these is to capture 42 channels of data.  Anyone know how this is done and how the boards are synchronized?

![IMG_4965](https://github.com/user-attachments/assets/425c7776-c93d-40b3-a4af-5fe1cd853448)

**Comment by gusmanb** [comment]
https://github.com/gusmanb/logicanalyzer/wiki/06---The-LogicAnalyzer-program#connecting-to-devices

https://github.com/gusmanb/logicanalyzer?tab=readme-ov-file#update-25022023---release-40-is-up-channels-a-go-go

**Reply by JimDrewGH** [reply]
Thank you!  Does this mean that we can simply plug two boards end to end?  Your link talks about 2 or 3 Dupont pins used for daisy chaining, but there is no info on how these are connected.

You might consider on the final release adding back the labeling on the silkscreen!  :)   I did figure out where R1 and R2 are located by looking at your prototype picture!

**Reply by gusmanb** [reply, marked as answer]
Yes, you can connect them directly, but the USB may collide depending on your cables, connection is straight 1:1.

These don't have labeling because I removed the pinouts that I added and printed them in labels to make them more readable and the component references where not needed for the manufacturer. When I publish the final files will be the KiCad project so you can export what you prefer.

If you whant when I get back to home I can send you the label poject files for Brother printers (the one I used).

**Reply by JimDrewGH** [reply]
It's ok.  I have a UV printer that can print on to the boards, and I really won't need this.  What I decided to do is make a micro USB hub for this.  The hub has dual USB micro-B male connectors that will plug directly into a pair of your boards that are stacked.  I think there will end up being 10mm of clearance between the boards to allow for some isolation (plastic) in the enclosure I am also designing.  I will use common standoffs to separate the boards, so the space between the boards will be determined by that.   The hub will have a single input on the opposite side, and maybe some hub status LEDs.  The case will need the labeling, so it won't matter for the PCB.  Having the boards stacked will keep it compact and I won't have to re-invent the Pico 2 layout for a (longer) single board solution.  :)

**Reply by gusmanb** [reply]
There are already solutions for this.
I have used this USB hub with very good results: https://es.aliexpress.com/item/1005005623071474.html

Is so small that it can be directly in the cable with a small printed case.

And if you are going to stack the boards remove the 3 pin male/female connectors and add in one of the boards a 3 pin socket on the top and on the other a 3 pin header on the bottom, in this  way they will connect when they get stacked.

**Comment by JimDrewGH** [comment]
I made a better solution for the hub.  It's all plug-in, no wiring.

![image](https://github.com/user-attachments/assets/e7ab771e-31e3-4cd4-b876-daf3259edf0a)

**Reply by gusmanb** [reply]
Next week I will add a functionality that you will use in the future, the capability of restarting the devices in BOOTPROG mode from software, to avoid the need of pressing the button to upgrade the firmware.

**Reply by JimDrewGH** [reply]
Your boards are stacked with 10mm standoffs, bolted to the bottom of the case.  Slots in the top and bottom of the case hold the hub board in place (so you can push and pull on the USB connector).  The top of the case is recessed for a label.  It uses 8mm round rubber feet.

I made the spacing between the boards so you can reach the button on the bottom board!

**Reply by JimDrewGH** [reply]
I will have PCBs early next week.  In the mean time... everything is ready, including the case.  Once tested I will make the files available.  BTW,  the latest Solidworks really upped their rendering game!
![LA1](https://github.com/user-attachments/assets/b415a279-ef3e-4cf3-a393-00300e7c09d0)
![LA6](https://github.com/user-attachments/assets/3d31010f-e114-4e23-898e-f3deebfe92d4)

**Comment by JimDrewGH** [comment]
Finally got around to building the logic analyzer hub.  These are not fun for the average hobbyist to build because the vertical USB connectors are through-hole with super tiny pins.  It works perfectly with the latest software!  I need to make a change to the case for the screw recesses on the bottom and then figure out the best way to release it.
![image1](https://github.com/user-attachments/assets/d990433d-7d64-408a-beb4-fcfbf953926b)
![image2](https://github.com/user-attachments/assets/0ec1b98d-3271-48db-a034-44479c4b5740)
![IMG_5038](https://github.com/user-attachments/assets/5233d4a2-0fdf-4d48-8e76-90507c947f2b)
![IMG_5039](https://github.com/user-attachments/assets/1f5d61ee-0c06-4733-bad9-af3a95cdabd7)
![IMG_5041](https://github.com/user-attachments/assets/83b17c69-f1c8-42ae-8958-74f8d130ee10)

**Comment by gusmanb** [comment]
Cool :)

I'm also preparing something so users that don't want to deal with assembling the hub can get it very easily.

![IMG_20241128_115825](https://github.com/user-attachments/assets/e76561df-f3ea-47c7-b7ed-8c35aeb4f649)

This is a test with two devices, but this hub supports up to 4 devices so depending on how many they stack they can add more or less outputs.

Also, I'm preparing a case for the hub with breakable holes, to break only the ones you need.

Finally I have created a little board that can be splitted to make the link between the analyzers.

![imagen](https://github.com/user-attachments/assets/cc61fd9e-a2a7-441e-90a9-1f93fdef9c5b)

BTW, the stacking idea is really great, it works very well, I'm thinking on removing the male pins and leave only the female for the link board.

**Comment by JimDrewGH** [comment]
I am using male connectors for the top board and female connectors for the bottom board.  You just plug the boards together with 10mm standoffs between them for a rigid connection between the boards.

My hub board uses the GL850G USB hub IC.  It's a 4 port hub IC, but I think 48 channels is plenty enough so I am only using 2 ports.  I could just increase the height of the board and case and add two more positions if needed.  Also, there is a pin compatible version of the hub IC, the GL852G which offers 4 individual (simultaneous) 480Mbps transfer tunnels.  This would be needed should the Pico 3 finally have high speed USB.

**Reply by gusmanb** [reply]
Yep, if you know exactly that you want only two boards using the vertical pins is optimal, but the boards I'm making are for unknown quantity, that's why I decided the link PCB can be a good option.

Regarding the hub, those are the same specs of the one that I have, it uses the FE1.1s, four 480Mbps inputs and four 480Mbps output streams. For a hobbyst that wants to assemble it itself is a very cheap and easy option.

BTW, I already have the implementation to put the devices in program mode from the software, but I will update the packages at least next week because I'm going to try to implement an autodetect function of the device instead of having to select manually the port, and in case of multidevice to avoid to know which port is which board.

**Comment by JimDrewGH** [comment]
I just realized that the connectors at both ends are connected together.  So, we really only need one pair of male/female connectors for joining the boards - not two!

**Reply by gusmanb** [reply]
With the stacking distribution, yes, my initial design was to cascade then and connect one board to the next one and up to five, that's why it has a pass-thru, you connect the top of one board to the bottom of the previous one.

**Reply by JimDrewGH** [reply]
This works out though because for stacking 4 boards total I just used male connectors on the bottom of the board and female connectors on the top of the board (opposite ends), and just extended my hub to include the two extra ports since it is 4 port hub.  This gives you 96 channels total, more importantly (for me at least) the ability to use 48 channels at a higher resolution because I would only be using 12 channel being captured per board, with the other channels ignored.

**Comment by gusmanb** [comment]
I have found a hub chip that looks really interesting: https://www.wch-ic.com/downloads/CH334DS1_PDF.html

The P variant is extremely small and to work it requires only three capacitors and a fuse... 

If you are going to redesign the hub to support the four ports it could be worth to try it.
I have ordered a sample hub that uses the chip to see how well performs, for my wired solution could be pretty cool as the hub board will be minuscule.

**Comment by JimDrewGH** [comment]
That's a great find!  The CH334P is definitely a simple design and it supports MTT!  I like the QFN parts as they are small and easy to work with.  We have problems getting the CH series chips in the U.S. though so I will look into availability and pricing.

**Comment by JimDrewGH** [comment]
The new USB hub design using the CH334P works perfectly!  There are several us in my developer group now using the logic analyzer in a stacked configuration with the case I designed.  The circuit board design (gerber files) you provided to me works perfectly, but of course you need to be able to solder fine pitch and 0402 size components.   This really is not something that your average DIY hobbyist can build.  There really is no reason for 0402 size parts.  There is plenty of space for 0603 (0805 even) size parts... but the buffers do require precision soldering (and also being perfectly clean as solder residue can throw off results).  The last picture below shows the harness I made with silicone wires and the ends with shrink tubing that is printed with a P-touch label machine.  This is the same setup that Saleae uses for their harnesses.

![la1](https://github.com/user-attachments/assets/a040ecb0-3378-4b27-963b-af2a16fcc65f)
![la2](https://github.com/user-attachments/assets/96b5170b-6231-44ba-b6e8-f72014d575c9)
![la3](https://github.com/user-attachments/assets/8b848dc3-78f1-4765-bb1b-a35764e67662)
![la4](https://github.com/user-attachments/assets/2db17ec5-efde-4866-9e6a-698929838d5f)

**Reply by noranraskin** [reply]
Would it be possible to exchange the USB in connector with a type C port?

**Reply by JimDrewGH** [reply]
Yes, that is not a bad idea!

**Comment by gpmidi** [comment]
@JimDrewGH Did you ever try to expand the USB hub side to a full stack of five?
